code/datums/controllers/process/gang.dm

```diff
@@ -0,0 +1,165 @@
+/datum/controller/process/gang_tag_vision
+	setup()
+		name = "Gang_Tags"
+		schedule_interval = GANG_TAG_SCAN_RATE
+
+	doWork()
+		for_by_tcl(tag, /obj/decal/gangtag)
+			tag.find_players()
+
+
+
+/datum/controller/process/gang_tag_score
+	setup()
+		name = "Gang_Tags_Score"
+		schedule_interval = GANG_TAG_SCORE_INTERVAL
+
+	doWork()
+		var/topHeat = 0
+		// calculate all heats
+		for_by_tcl(I, /obj/decal/gangtag)
+			topHeat = max(I.calculate_heat(), topHeat)
+
+		for_by_tcl(I, /obj/decal/gangtag)
+			if (!I || I.disposed || I.qdeled)
+				continue
```

Comment:
```suggestion
```
Was marked as resolved, dunno if missed? Think it could still be removed

Reply:
yeah i may have missed a couple. vscode's git integ started messing up real bad later into it